src/Services/LightningService.cs

```diff
@@ -249,11 +249,74 @@ public async Task OpenChannel(ChannelOperationRequest channelOperationRequest)
                     NodePubkey = ByteString.CopyFrom(Convert.FromHexString(destination.PubKey)),
                 };
 
+                //Prior to opening the channel, we add the remote node as a peer
+                var remoteNodeInfo = await GetNodeInfo(channelOperationRequest.DestNode?.PubKey);
+                if (remoteNodeInfo == null)
+                {
+                    _logger.LogError("Error, remote node with {Pubkey} not found",
+                        channelOperationRequest.DestNode?.PubKey);
+                    throw new InvalidOperationException();
+                }
+
+                //For now, we only rely on pure tcp IPV4 connections
+                var addr = remoteNodeInfo.Addresses.FirstOrDefault(x => x.Network == "tcp").Addr;
+                
+                if(addr == null)
+                {
+                    _logger.LogError("Error, remote node with {Pubkey} has no tcp IPV4 address",
+                        channelOperationRequest.DestNode?.PubKey);
+                    throw new InvalidOperationException();
+                }
+                var isPeerAlreadyConnected = false;
+
+                ConnectPeerResponse connectPeerResponse = null;
+                try
+                {
+                    connectPeerResponse = await client.ConnectPeerAsync(new ConnectPeerRequest
+                    {
+                        Addr = new LightningAddress {Host = addr, Pubkey = remoteNodeInfo.PubKey},
+                        Perm = true
+                    }, new Metadata
+                    {
+                        {"macaroon", source.ChannelAdminMacaroon}
+                    });
+                }
+                //We avoid to stop the method if the peer is already connected
+                catch (RpcException e)
+                {
+                    if (!e.Message.Contains("already connected to peer"))
+                    {
+                        throw;
+                    }
+                    else
+                    {
+                        isPeerAlreadyConnected = true;
+                    }
```

Comment:
what if the rpc error is that it couldn't establish the connection?

Reply:
This is a try-catch inside of the big try-catch block, so any exception will pause the job and report the error, the thing about this specific try-catch is to handle a exception which should not be throw (peer already connected)